#! /bin/sh /usr/share/dpatch/dpatch-run
## ldap-subtree.dpatch by Christoph Haas <haas@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix LDAP search query
## DP: http://osdir.com/ml/network.dns.powerdns.devel/2008-03/msg00004.html
## DP: http://wiki.powerdns.com/cgi-bin/trac.fcgi/changeset?new=trunk%2Fpdns%2Fmodules%2Fldapbackend%2Fldapbackend.cc%401152&old=trunk%2Fpdns%2Fmodules%2Fldapbackend%2Fldapbackend.cc%401107

@DPATCH@
diff -urNad pdns~/modules/ldapbackend/ldapbackend.cc pdns/modules/ldapbackend/ldapbackend.cc
--- pdns~/modules/ldapbackend/ldapbackend.cc	2008-11-25 21:22:42.000000000 +0100
+++ pdns/modules/ldapbackend/ldapbackend.cc	2008-11-25 22:18:04.000000000 +0100
@@ -127,7 +127,7 @@
 	qesc = toLower( m_pldap->escape( target ) );
 
 	// search for SOARecord of target
-	filter = strbind( ":target:", "associatedDomain=" + qesc, getArg( "filter-axfr" ) );
+    filter = strbind( ":target:", "&(associatedDomain=" + qesc + ")(sOARecord=*)", getArg( "filter-axfr" ) );
 	m_msgid = m_pldap->search( dn, LDAP_SCOPE_SUBTREE, filter, (const char**) ldap_attrany );
 	m_pldap->getSearchEntry( m_msgid, m_result, true );
 
